### PR TITLE
Added Decodable

### PR DESCRIPTION
**Pull request by alessandroorru**
I added [Decodable framework](https://github.com/Anviking/Decodable).

In the README I just added the link and the snippet of code used, but I leave up to you to update from the Analysis section forward.

### CODE REVIEW

**Comment by codecaffeine** [comment]
Great work. According to the tests it performed somewhere between Marshall and Mapper. I’m guessing because it’s more architecturally similar to Marshall but it has custom operators?

**Comment by alessandroorru** [comment]
@codecaffeine Unfortunately I'm not an expert of the implementation of both frameworks, anyway I think that both of them don't use reflection as the way to go, preferring explicit types and decoding. 

I'm using Decodable because I like its simple syntax (compared to Argo) and its speed, that was great at the time I had to implement it. The operator is actually just one (plus an optional variant) and it's straightforward to use. Here I just wanted to test it out to see if it is still good in terms of performance compared to newer libraries, and it seems it doesn't need to be replaced ;)

**Comment by codecaffeine** [comment]
@alessandroorru yeah, I’m using Decodable as well. I tried to roll my own parser with reflection but it was definitely slow and fragile. I’m happy with these performance numbers since I prefer the syntax to that of Marshall.

Even though there are only a few operators, there are a 163 overloads for the operators (https://github.com/Anviking/Decodable/blob/master/Sources/Overloads.swift), but I’m not a swift performance expert so I’m not sure if that would affect the performance.

**Comment by alessandroorru** [comment]
@codecaffeine I think they could have an impact, since they are all generics, anyway from what I read [here](https://swift.org/blog/whole-module-optimizations) turning on whole module optimization could highly enhance the performance

File: JSONShootout.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -73,6 +82,14 @@
 /* End PBXCopyFilesBuildPhase section */
 
 /* Begin PBXFileReference section */
+		5BAB43C51DBE06A90017C770 /* Mapper.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = Mapper.framework; path = "../../Library/Developer/Xcode/DerivedData/JSONShootout-bkocarfdpnsffvahaikdrlrmawsu/Build/Products/Debug-iphonesimulator/Mapper.framework"; sourceTree = "<group>"; };
+		5BAB43C61DBE06A90017C770 /* Marshal.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = Marshal.framework; path = "../../Library/Developer/Xcode/DerivedData/JSONShootout-bkocarfdpnsffvahaikdrlrmawsu/Build/Products/Debug-iphonesimulator/Marshal.framework"; sourceTree = "<group>"; };
```
Comment (bwhiteley) [inline JSONShootout.xcodeproj/project.pbxproj]: Note that the existing frameworks have been changed to refer to user-specific derived data paths.

File: JSONShootout.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -73,6 +82,14 @@
...
+		5BAB43C81DBE06A90017C770 /* Unbox.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = Unbox.framework; path = "../../Library/Developer/Xcode/DerivedData/JSONShootout-bkocarfdpnsffvahaikdrlrmawsu/Build/Products/Debug-iphonesimulator/Unbox.framework"; sourceTree = "<group>"; };
+		5BAB43E21DBE07D50017C770 /* Decodable.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = Decodable.framework; path = "Carthage/Checkouts/Decodable/build/Debug-iphoneos/Decodable.framework"; sourceTree = "<group>"; };
+		5BAB43E41DBE07DD0017C770 /* Decodable.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = Decodable.framework; path = "Carthage/Checkouts/Decodable/build/Debug-iphoneos/Decodable.framework"; sourceTree = "<group>"; };
+		5BAB43E61DBE07E10017C770 /* Decodable.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = Decodable.framework; path = "Carthage/Checkouts/Decodable/build/Debug-iphoneos/Decodable.framework"; sourceTree = "<group>"; };
```
Comment (bwhiteley) [inline JSONShootout.xcodeproj/project.pbxproj]: Carthage paths should not be here.

File: JSONShootout.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -519,8 +569,12 @@
 		CC5A1A1F1CEADE59007238F3 /* Debug */ = {
 			isa = XCBuildConfiguration;
 			buildSettings = {
+				FRAMEWORK_SEARCH_PATHS = (
+					"$(inherited)",
+					"$(PROJECT_DIR)/Carthage/Checkouts/Decodable/build/Debug-iphoneos",
+				);
 				INFOPLIST_FILE = JSONShootoutTests/Info.plist;
-				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks @loader_path/Frameworks";
+				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks @loader_path/Frameworks $(FRAMEWORK_SEARCH_PATHS) $(PROJECT_DIR)/Carthage/Build/iOS";
```
Comment (bwhiteley) [inline JSONShootout.xcodeproj/project.pbxproj]: This carthage path should not be here.

**Comment by bwhiteley** [comment]
Thanks for the PR. Please see the Contributing notes at the bottom of the Readme.

**Comment by alessandroorru** [comment]
Ops sorry, didn't notice the contributing notes, I added the frameworks and the search path because when I first tested it out I wasn't able to compile it. I will fix them as noted and push.

**Comment by bwhiteley** [comment]
Note that I also split each extension into its own file. Sorry for the conflicts.

**Comment by alessandroorru** [comment]
@bwhiteley It's ok, it was trivial. I updated it, should be fine now.

However, in order to be able to run the tests I had to add ModelObjects.frameworks to the linked libraries of JSONShootoutTests, otherwise compile was fine but running tests produced just `Test target X encountered an error (Early unexpected exit, operation never finished bootstrapping - no restart will be attempted`.
In any case, that was the only edit in the [last commit](https://github.com/bwhiteley/JSONShootout/pull/7/commits/5b14b27cb1e4c5b2a8f82feab1437d45e612759f)

Check it out and let me know if now it is ok

**Pull request merged**